base/io.jl

```diff
@@ -167,24 +167,50 @@ The text is assumed to be encoded in UTF-8.
 readuntil(filename::AbstractString, args...) = open(io->readuntil(io, args...), filename)
 
 """
-    readline(stream::IO=STDIN)
-    readline(filename::AbstractString)
+    readline(stream::IO=STDIN, chomp::Bool=false)
+    readline(filename::AbstractString, chomp::Bool=false)
```

Comment:
I wonder if `chomp=true` should be the default (here and in `eachline`).  Almost all uses of `readline` that I can find in the packages either (a) are insensitive to whether it ends with a newline or (b) explicitly strip the newline.

That would be a breaking change, but I doubt it would break much actual code, since most code calling `readline` or `eachline` has to handle the case where the last line in the file does not end in a newline anyway.